examples/protocol_specification_ex/ml_trade.yaml

```diff
@@ -29,3 +29,12 @@ ct:Query: |
 ct:Description: |
   bytes description = 1;
 ...
+---
+reply:
+  cfp: [terms]
+  terms: [accept]
+  accept: [data]
+  data: [accept, terms]
+roles: {seller, buyer}
+end_states: [successful, open]
```

Comment:
As above. A _terminated_ `ml_trade` cannot be `open`, right? Is it always successful once terminated? Also when does it terminate? From the reply structure it seems it can be infinite. How do we know when an `ml_trade` dialogue is terminated?

Reply:
I think it should terminate with `data`. You can't receive/send other messages after `data`.

Reply:
I think `data` needs to be fixed to be `data: []`. And yes, `end_states: [closed]` is the only one.

Reply:
I agree with the change to replies for `data`. In line with my comments above, I suggest something else over `close`, maybe `data_exchanged` or something.

Incidentally, this dialogue seems it always succeeds (ends with data being exchanged). So with this dialogue definition, agents can't really decline terms. It is fine if its purpose is only to demonstrate a use-case.